src/games/hex_state.h

```diff
@@ -80,14 +80,16 @@ void Hex::State<SIZE, PIE>::ApplyAction(const _Action& action) {
   // find board move from action
   int i = action.GetY();
   int j = action.GetZ();
+  assert(i >= 0 && i < SIZE);  // N added
+  assert(j >= 0 && j < SIZE);
   int index = _board.convertCellToIndex(Cell(i, j));
   std::optional<int> lastIndex = _board.getLastIndex();
 
   // update features
   // TODO assert action is in legal actions ?
   if (not lastIndex or *lastIndex != index) {
     Color currentColor = _board.getCurrentColor();
-    _features.at(((currentColor * SIZE) + i) * SIZE + j) = 1.f;
+    _features.at((((currentColor) * SIZE) + i) * SIZE + j) = 1.f;
```

Comment:
strange extra parens?
```suggestion
    _features.at(((currentColor * SIZE) + i) * SIZE + j) = 1.f;
```